src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -20,6 +21,78 @@ private JsonDynamicObject(Dictionary<JsonProperty, JsonValue> properties)
             _properties = properties;
         }
 
+        private static Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>> TypeCache = new Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>>();
+
+        public static T Deserialize<T>(ReadOnlySpan<byte> utf8)
+        {
+            if (!TypeCache.TryGetValue(typeof(T), out Dictionary<int, PropertyInfoLinkedList> dictionary))
+            {
+                dictionary = GetTypeMap(typeof(T));
+                TypeCache.Add(typeof(T), dictionary);
+            }
+
+            T instance = Deserializer<T>.Deserialize(utf8);
```

Comment:
BTW, the name of the method above is very confusing. It's not actually deserializing the whole thing; it's just creating the top level instance. correct?

Reply:
Yes, for now. I will change the method name. The intention is to have the entire deserialize method (the reader loop) by emitted in IL.